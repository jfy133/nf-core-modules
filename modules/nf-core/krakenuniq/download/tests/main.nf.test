// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test krakenuniq/download
nextflow_process {

    name "Test Process KRAKENUNIQ_DOWNLOAD"
    script "../main.nf"
    process "KRAKENUNIQ_DOWNLOAD"

    tag "modules"
    tag "modules_nfcore"
    tag "krakenuniq"
    tag "krakenuniq/download"

    test("taxonomy") {

        when {
            process {
                """
                input[0] = 'taxonomy'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("taxonomy - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = 'taxonomy'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
