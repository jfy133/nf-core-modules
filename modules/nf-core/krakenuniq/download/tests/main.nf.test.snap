{
    "taxonomy - stub": {
        "content": [
            {
                "0": [
                    [
                        
                    ]
                ],
                "1": [
                    "versions.yml:md5,a12e821f0d977f77c0c905a89c708d09"
                ],
                "output": [
                    [
                        
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a12e821f0d977f77c0c905a89c708d09"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-01-17T08:39:20.362801897"
    },
    "taxonomy": {
        "content": [
            {
                "0": [
                    [
                        "names.dmp:md5,aae05a99d67ea8a593f8642b40e98f28",
                        "nodes.dmp:md5,73f515a0c4be7a324a6717bbe14cd274",
                        "taxdump.tar.gz:md5,4ec3d4430ef2d350db5b3284edd33a8b",
                        "timestamp:md5,6a90d10ebd6398165a657c52402468ee"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a12e821f0d977f77c0c905a89c708d09"
                ],
                "output": [
                    [
                        "names.dmp:md5,aae05a99d67ea8a593f8642b40e98f28",
                        "nodes.dmp:md5,73f515a0c4be7a324a6717bbe14cd274",
                        "taxdump.tar.gz:md5,4ec3d4430ef2d350db5b3284edd33a8b",
                        "timestamp:md5,6a90d10ebd6398165a657c52402468ee"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a12e821f0d977f77c0c905a89c708d09"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-01-17T08:39:02.992235854"
    }
}